lldb/source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp

```diff
@@ -1125,9 +1125,8 @@ bool SwiftASTManipulator::AddExternalVariables(
 
     redirected_var_decl->setImplicit(true);
 
-    m_source_file.Decls.insert(m_source_file.Decls.begin(), top_level_code);
-    m_source_file.Decls.insert(m_source_file.Decls.begin(),
-                               redirected_var_decl);
+    m_source_file.addTopLevelDecl(redirected_var_decl);
```

Comment:
@dcci Do you know why this is injecting variables into the *start* of the top level decls list instead of the end?

Reply:
Quite honestly, I don't. This predates my time. Do you know what breaks when you change it? I'll be happy to take a look with you.

Reply:
Quite honestly, I don't. This predates my time. Do you know what breaks when you change it? I'll be happy to take a look with you.

Reply:
If I put these in the original order (Redirected VarDecl then Top Level Code with Pattern Binding) then it fails a REPL test.  But the odd thing is that it should have failed much *much* more spectacularly than just a runtime exclusivity assertion.  

The correct way around (Top Level Code then Redirected VarDecl, we'll see soon enough...

Reply:
Thanks. I'm curious to see the results too. @jimingham might have more context on why this was the order originally, but it was long enough ago that he might not remember either and the history is lost in the initial import.

Reply:
Thanks. I'm curious to see the results too. @jimingham might have more context on why this was the order originally, but it was long enough ago that he might not remember either and the history is lost in the initial import.

Reply:
Same REPL test fails.  I can merge this through with an extra entrypoint on the Swift side to preserve this behavior, but the fact that only a single test fails because of this, and fails in such a bizarre way, suggests there's something deeper at play.